### PR TITLE
SJQ2T-195: BE Endpoint For Delete Bucket

### PR DESCRIPTION
**Pull request by DarkoVeljanoskiHtec**
added endpoint in the controller, implemented the service and added two tests as well.

### CODE REVIEW

File: backend/src/main/java/com/htecgroup/skynest/service/BucketService.java [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+import com.htecgroup.skynest.model.dto.BucketDto;
 import com.htecgroup.skynest.model.response.BucketResponse;
 
 import java.util.UUID;
 
 public interface BucketService {
 
   BucketResponse getBucket(UUID uuid);
+
+  BucketDto findBucketById(UUID uuid);
```
Comment (petarstoimenovskihtec) [inline backend/src/main/java/com/htecgroup/skynest/service/BucketService.java:15]: What is the correct returned mode? you have BucketResponse and BucketDto ? try to be consistent as in the other services

Comment (DarkoVeljanoskiHtec) [reply backend/src/main/java/com/htecgroup/skynest/service/BucketService.java:15]: BucketResponse is the correct one who is connected to the controller, but I use BucketDto for the delete implementation and for easier tests, the same thing is done in the user service as well.

File: backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java [hunk 2]
```diff
@@ -97,4 +94,54 @@ public ResponseEntity<BucketResponse> getBucket(@PathVariable UUID uuid) {
...
+                    @ExampleObject(
+                        value =
+                            "{\"messages\":[\"Bucket is already disabled.\"],"
+                                + " \"status\": \"409\","
+                                + " \"timestamp\": \"2022-06-07 16:18:12\"}")
+                  })
+            })
+      })
+  @PutMapping("delete/{uuid}")
+  public ResponseEntity<Boolean> deleteBucket(@PathVariable UUID uuid) {
```
Comment (ivanfhtec) [inline backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java:231]: can this return 401?

File: backend/src/main/java/com/htecgroup/skynest/service/BucketService.java (hunk 1 above)
Comment (ivanfhtec) [reply backend/src/main/java/com/htecgroup/skynest/service/BucketService.java:15]: Could we use BucketEntity here instead then?

Other services wouldn't have to import bucket repo

More refactoring

File: backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java (hunk 2 above)
Comment (DarkoVeljanoskiHtec) [reply backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java:231]: We don't have preAuthorize yet so i guess not.

Comment (ivanfhtec) [reply backend/src/main/java/com/htecgroup/skynest/controller/BucketController.java:231]: invalid jwt? this endpoint requires a token, I think

**Review by ivanfhtec — APPROVED** [review]

**Pull request merged**